src/Core/NotificationCenter/Repositories/INotificationRepository.cs

```diff
@@ -0,0 +1,42 @@
+﻿#nullable enable
+using Bit.Core.NotificationCenter.Entities;
+using Bit.Core.NotificationCenter.Enums;
+using Bit.Core.NotificationCenter.Models.Filter;
+using Bit.Core.Repositories;
+
+namespace Bit.Core.NotificationCenter.Repositories;
+
+public interface INotificationRepository : IRepository<Notification, Guid>
+{
+    /// <summary>
+    /// Get notifications for a user with the given filter.
+    /// Includes global notifications and notifications with status.
+    /// </summary>
+    /// <param name="userId">User Id</param>
+    /// <param name="clientType">
+    /// Filter for notifications by client type. Always includes notifications with <see cref="ClientType.All"/>.
+    /// </param>
+    /// <returns>
+    /// Ordered by priority (highest to lowest) and creation date (descending).
+    /// </returns>
+    Task<IEnumerable<Notification>> GetByUserIdAsync(Guid userId, ClientType clientType);
```

Comment:
❓ Is is worth maintaining two methods here, or one with an optional status?

Reply:
Fixed in Fixed in 006f1b4